### PR TITLE
New depot system

### PR DESCRIPTION
**Pull request by comedinha**
Initial new depot system
Pagination have 1 little problem - When you chanche the page not is possible back to depot chest (is protocolgame problem? - only need open the chest again)
If the server use depotid 0 he go to town 17, if you server have > 17 towns, all items go to town 17.

### CODE REVIEW

File: src/container.cpp [hunk 1]
```diff
@@ -306,7 +306,7 @@ ReturnValue Container::queryAdd(int32_t index, const Thing& thing, uint32_t coun
 	}
 
 	const Cylinder* cylinder = getParent();
-	if (!hasBitSet(FLAG_NOLIMIT, flags)) {
+	if (!hasBitSet(FLAG_NOLIMIT, flags) || !hasPagination()) {
```
Comment (slavidodo) [inline src/container.cpp]: exactly, the point where I didn't notify that mark had done a problem in!

File: data/items/items.xml [hunk 2]
```diff
@@ -30242,6 +30242,57 @@
...
+		<attribute key="containerSize" value="32" />
+	</item>
+	<item id="25467" article="a" name="depot box XV">
+		<attribute key="containerSize" value="32" />
+	</item>
+	<item id="25468" article="a" name="depot box XVI">
+		<attribute key="containerSize" value="32" />
+	</item>
+	<item id="25469" article="a" name="depot box XVII">
+		<attribute key="containerSize" value="32" />
```
Comment (slavidodo) [inline data/items/items.xml:30294]: Should you have set container size here?

in depotchest.cpp

``` c++
maxSize = 32;
```

File: src/container.cpp (hunk 1 above)
Comment (comedinha) [inline src/container.cpp]: ok, I will test

File: data/movements/scripts/tiles.lua [hunk 3]
```diff
@@ -27,7 +27,10 @@ function onStepIn(creature, item, position, fromPosition)
 		lookPosition:getNextPosition(player:getDirection())
 		local depotItem = Tile(lookPosition):getItemByType(ITEM_TYPE_DEPOT)
 		if depotItem ~= nil then
-			local depotItems = player:getDepotChest(getDepotId(depotItem:getUniqueId()), true):getItemHoldingCount()
+			local depotItems = 0
+			for i = 0, 16 do
```
Comment (vankk) [inline data/movements/scripts/tiles.lua]: 1-17? :p

**Comment by slavidodo** [comment]
I am not sure about it, should your inbox work?

If not, I 've already fixed my depot system and now should work for everything.

**Comment by vankk** [comment]
> I am not sure about it, should your inbox work?
> 
> If not, I 've already fixed my depot system and now should work for everything.

It's working fine!!

**Comment by comedinha** [comment]
"squash" on https://github.com/Tulioh/forgottenserver/pull/3

**Pull request closed without merging**